compiler/src/dotty/tools/dotc/core/Types.scala

```diff
@@ -435,6 +425,13 @@ object Types {
         val rsym = r.classSymbol
         if (lsym isSubClass rsym) rsym
         else if (rsym isSubClass lsym) lsym
+        else if (ctx.explicitNulls && this.isNullableUnion) {
+          val OrType(left, _) = this.normNullableUnion
```

Comment:
`classSymbol` needs to be fast, and ``isNullableUnion` and normNullableUnion` is slow. So we have to fix that. It should be possible to define this method without reference to either.

Reply:
This code turned out to no be necessary. Done.